pkg/kubelet/cm/cpumanager/topology_hints_test.go

```diff
@@ -254,6 +257,160 @@ func TestGetPodTopologyHints(t *testing.T) {
+					Preferred:        false,
+				},
+				{
+					NUMANodeAffinity: m1011,
+					Preferred:        false,
+				},
+				{
+					NUMANodeAffinity: m1101,
+					Preferred:        false,
+				},
+				{
+					NUMANodeAffinity: m1111,
+					Preferred:        false,
+				},
+			},
+		},
+		{
+			// CPU available on numa node[0 ,1]. CPU on numa node 0 can satisfy request of 2 CPU's
+			description:   "AlignBySocket:true Preferred hints contains socket aligned hints",
+			pod:           *testPod1,
+			container:     *testContainer1,
+			defaultCPUSet: cpuset.NewCPUSet(2, 3, 11),
+			topology:      topoDualSocketMultiNumaPerSocketHT,
+			policyOptions: map[string]string{AlignBySocketOption: "true"},
+			expectedHints: []topologymanager.TopologyHint{
+				{
+					NUMANodeAffinity: m0001,
+					Preferred:        true,
+				},
+				{
+					NUMANodeAffinity: m0011,
+					Preferred:        true,
+				},
+				{
+					NUMANodeAffinity: m0101,
+					Preferred:        false,
+				},
+				{
+					NUMANodeAffinity: m1001,
+					Preferred:        false,
+				},
+				{
+					NUMANodeAffinity: m0111,
+					Preferred:        false,
+				},
+				{
+					NUMANodeAffinity: m1011,
+					Preferred:        false,
+				},
+				{
+					NUMANodeAffinity: m1101,
+					Preferred:        false,
+				},
+				{
+					NUMANodeAffinity: m1111,
+					Preferred:        false,
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			var activePods []*v1.Pod
+			for p := range testCase.assignments {
+				pod := v1.Pod{}
+				pod.UID = types.UID(p)
+				for c := range testCase.assignments[p] {
+					container := v1.Container{}
+					container.Name = c
+					pod.Spec.Containers = append(pod.Spec.Containers, container)
+				}
+				activePods = append(activePods, &pod)
+			}
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, pkgfeatures.CPUManagerPolicyAlphaOptions, true)()
+			policyOpt, _ := NewStaticPolicyOptions(testCase.policyOptions)
+			m := manager{
+				policy: &staticPolicy{
+					topology: testCase.topology,
+					options:  policyOpt,
+				},
+				state: &mockState{
+					assignments:   testCase.assignments,
+					defaultCPUSet: testCase.defaultCPUSet,
+				},
+				topology:          testCase.topology,
+				activePods:        func() []*v1.Pod { return activePods },
+				podStatusProvider: mockPodStatusProvider{},
+				sourcesReady:      &sourcesReadyStub{},
+			}
+
+			podHints := m.GetPodTopologyHints(&testCase.pod)[string(v1.ResourceCPU)]
+			sort.SliceStable(podHints, func(i, j int) bool {
+				return podHints[i].LessThan(podHints[j])
+			})
+			sort.SliceStable(testCase.expectedHints, func(i, j int) bool {
+				return testCase.expectedHints[i].LessThan(testCase.expectedHints[j])
+			})
+			if !reflect.DeepEqual(testCase.expectedHints, podHints) {
+				t.Errorf("Expected in result to be %v , got %v", testCase.expectedHints, podHints)
+			}
+		})
+	}
+}
+
```

Comment:
At some point it would be good to add unit tests in `pkg/kubelet/cm` that can test across all of the hint-providers (with their various options) instead of just testing each of them in isolation like this.

Reply:
Will track this item for beta. 
I think testcases under pkg/kubelet/cm are definitely required and should be tracked via separate issue